src/test/groovy/unit/org/pih/warehouse/requisition/RequisitionItemSpec.groovy

```diff
@@ -13,88 +16,346 @@ import spock.lang.Unroll
+
+        then:
+        calculatedTotalQuantityNotCanceled == totalNotCanceledQuantity
+
+        where:
+        notCanceledQuantity | productPackageQuantity || totalNotCanceledQuantity
+        null                | null                   || 0
+        1                   | null                   || 1
+        null                | 1                      || 0
+        10                  | 5                      || 50
+    }
+
+    void "RequisitionItem.isCanceled() should return: #isCanceled when #quantity and #quantityCanceled and #modificationItem and #substitutionItem and #requisitionItems are passed"() {
+        given:
+        RequisitionItem requisitionItem = Spy(RequisitionItem) {
+            totalQuantityCanceled() >> quantityCanceled
+            totalQuantity() >> quantity
+        }
+
+        requisitionItem.modificationItem >> modificationItem
+        requisitionItem.substitutionItem >> substitutionItem
+        requisitionItem.requisitionItems >> requisitionItems
+
+        when:
+        Boolean isRequisitionCanceled = requisitionItem.isCanceled()
+
+        then:
+        isCanceled == isRequisitionCanceled
+
+        where:
+        quantityCanceled | quantity | modificationItem      | substitutionItem      | requisitionItems        || isCanceled
+        3                | 3        | null                  | null                  | null                    || true
+        3                | 2        | null                  | null                  | null                    || false
+        3                | 2        | Mock(RequisitionItem) | null                  | null                    || false
+        3                | 2        | null                  | Mock(RequisitionItem) | null                    || false
+        3                | 2        | null                  | null                  | [Mock(RequisitionItem)] || false
+    }
+
+    void "RequisitionItem.isCanceledDuringPick() should return: #isCanceled when requisition status is #status and quantityPicked of modificationItem is #modificationItemQuantityPicked"() {
+        given:
+        RequisitionItem modificationItem = Stub(RequisitionItem) {
+            calculateQuantityPicked() >> modificationItemQuantityPicked
+        }
+
+        domain.modificationItem = modificationItem
+        domain.requisition.status = status
+
+        when:
+        Boolean isCanceledDuringPick = domain.isCanceledDuringPick()
+
+        then:
+        isCanceledDuringPick == isCanceled
+
+        where:
+        status                      | modificationItemQuantityPicked || isCanceled
+        RequisitionStatus.PICKING   | 0                              || true
+        RequisitionStatus.ISSUED    | 0                              || true
+        RequisitionStatus.VERIFYING | null                           || false
+        RequisitionStatus.PICKING   | null                           || false
```

Comment:
could be nice to add some test cases around the modificationItem being set since I see that's used in the isCanceledDuringPick() method. Maybe something similar to what you did for the last test you have in this file. Up to you though.